rholang/src/test/scala/coop/rchain/rholang/interpreter/matcher/MatchTest.scala

```diff
@@ -922,12 +925,24 @@ class VarMatcherSpec extends FlatSpec with Matchers with TimeLimits with TripleE
     assertSpatialMatch(failTarget, pattern, None)
   }
 
-  "Spatial matcher" should "short-circuit when runs out of phlo in the middle of matching" in {
+  "spatialMatch" should "short-circuit when runs out of phlo in the middle of matching" in {
     val target: Par = EList(Seq(GInt(1), GInt(2), GInt(3)))
     val pattern: Par =
       EList(Seq(GInt(1), EVar(FreeVar(0)), EVar(FreeVar(1))), connectiveUsed = true)
     val res =
       spatialMatch[NonDetFreeMapWithCost, Par, Par](target, pattern).runFirstWithCost(Cost(0))
     res should be(Left(OutOfPhlogistonsError))
   }
+
+  "spatialMatchAndCharge" should "short-circuit when runs out of phlo in the middle of matching" in {
+    implicit val cost: _cost[Task] = CostAccounting.unsafe[Task](Cost(0))
+
+    val target: Par = EList(Seq(GInt(1), GInt(2), GInt(3)))
+    val pattern: Par =
+      EList(Seq(GInt(1), EVar(FreeVar(0)), EVar(FreeVar(1))), connectiveUsed = true)
```

Comment:
Whelp... Ideally we'd somehow share this copied test data, but won't insist on that :)